go-controller/pkg/ovn/controller/egress_services/egress_services_controller.go

```diff
@@ -417,16 +490,337 @@ func (c *Controller) repair() error {
+		// selecting a new node for it.
+		return c.clearServiceResources(key, state)
+	}
+
+	// At this point the states are valid and we should create the proper logical router policies.
+	// We reach the desired state by fetching all of the endpoints associated to the service and comparing
+	// to the known state:
+	// We need to create policies for endpoints that were fetched but not found in the cache,
+	// and delete the policies for those which are found in the cache but were not fetched.
+	// We do it in one transaction, if it succeeds we update the cache to reflect the new state.
+
+	v4ToAdd := v4Endpoints.Difference(state.v4Endpoints).UnsortedList()
+	v6ToAdd := v6Endpoints.Difference(state.v6Endpoints).UnsortedList()
+	v4ToRemove := state.v4Endpoints.Difference(v4Endpoints).UnsortedList()
+	v6ToRemove := state.v6Endpoints.Difference(v6Endpoints).UnsortedList()
+
+	allOps := []libovsdb.Operation{}
+	createOps, err := c.createLogicalRouterPoliciesOps(key, node.v4MgmtIP.String(), node.v6MgmtIP.String(), v4ToAdd, v6ToAdd)
+	if err != nil {
+		return err
+	}
+	allOps = append(allOps, createOps...)
+
+	createOps, err = c.addPodIPsToAddressSetOps(createIPAddressNetSlice(v4ToAdd, v6ToAdd))
+	if err != nil {
+		return err
+	}
+	allOps = append(allOps, createOps...)
+
+	deleteOps, err := c.deleteLogicalRouterPoliciesOps(key, v4ToRemove, v6ToRemove)
+	if err != nil {
+		return err
+	}
+	allOps = append(allOps, deleteOps...)
+
+	deleteOps, err = c.deletePodIPsFromAddressSetOps(createIPAddressNetSlice(v4ToRemove, v6ToRemove))
+	if err != nil {
+		return err
+	}
+	allOps = append(allOps, deleteOps...)
+
+	if _, err := libovsdbops.TransactAndCheck(c.nbClient, allOps); err != nil {
+		return fmt.Errorf("failed to update router policies for %s, err: %v", key, err)
+	}
+
+	state.v4Endpoints.Insert(v4ToAdd...)
+	state.v4Endpoints.Delete(v4ToRemove...)
+	state.v6Endpoints.Insert(v6ToAdd...)
+	state.v6Endpoints.Delete(v6ToRemove...)
+
+	// We configured OVN - the last step is to label the node and set the status
+	// to mark it as the node holding the service.
+
+	err = c.setEgressServiceHost(namespace, name, state.node) // set the EgressService status, will also override manual changes
+	if err != nil {
+		return err
+	}
+
+	return c.labelNodeForService(namespace, name, node.name)
+}
+
+// Removes all of the resources that belong to the egress service.
+// This includes removing the host status value, the logical router policies,
+// the label from the node and updating the caches.
+// This also requeues the service after cleaning up to be sure we are not
+// missing an event after marking it as stale that should be handled.
+// This should only be called with the controller locked.
+func (c *Controller) clearServiceResources(key string, svcState *svcState) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	svcState.stale = true
+	if err := c.setEgressServiceHost(namespace, name, ""); err != nil {
```

Comment:
I think that clearing the status should be done at the same time as removing the node label, same as: https://github.com/ovn-org/ovn-kubernetes/pull/3389#discussion_r1163076492

Reply:
done (both happen after the lrps now)